### PR TITLE
Replace ema with long term latency in average-latency blueprint

### PR DESCRIPTION
**Pull request by hasit**
### Description of change

screencapture-localhost-3000-d-1cafa24eadc87e0eeb9217b9179e31473bcfa21d-aperture-service-protection-2023-06-27-13_28_33

##### Checklist

- [x] Tested in playground or other setup
- [x] Screenshot (Grafana) from playground added to PR for 15+ minute run
- [x] Documentation is changed or added
- [ ] Tests and/or benchmarks are included
- [ ] Breaking changes

### Summary by OpenAI

**New Feature:**
- Replaced EMA latency calculation with a long-term average latency calculation in the `average-latency` blueprint.
- Added two new parameters for the long-term query interval and periodic interval.
- Changed the tolerance factor from 2.0 to 1.25.

**Chore:**
- Updated the version of a dependency across multiple scenarios in the playground.

> 🎉 With every tick and tock of the clock,  
> Our code evolves, solid as a rock.  
> Latency measured with a new trick,  
> And dependencies updated quick.  
> Celebrate this change, let's pop the cork,  
> For this is how we shape our work! 🥂🚀

### Summary by CodeRabbit

New Feature: The `average-latency` blueprint of service-protection policies has been updated to use a long-term average latency calculation, with new parameters for query interval and periodic interval. The tolerance factor has also been changed from 2.0 to 1.25. This PR also includes an update to the `k6-utils` library version in some load-generator tests.

> "Latency calculations refined,
> Load generators now aligned.
> CodeRabbit's review so fine,
> Quality improved, line by line."

### CODE REVIEW

File: docs/content/reference/blueprints/policies/service-protection/average-latency.md [hunk 1]
```diff
@@ -204,42 +204,42 @@
...
-    name='policy.latency_baseliner.latency_tolerance_multiplier'
-    description='Tolerance factor beyond which the service is considered to be in overloaded state. E.g. if EMA of latency is 50ms and if Tolerance is 1.1, then service is considered to be in overloaded state if current latency is more than 55ms.'
-    type='Number (double)'
+    name='policy.latency_baseliner.long_term_query_periodic_interval'
+    description='Periodic interval for long-term latency query, i.e. how often the query is run.'
+    type='string'
     reference=''
-    value='"__REQUIRED_FIELD__"'
+    value='"30s"'
 />
```
[the remark this answers is not in the document]
Comment (hasit) [reply docs/content/reference/blueprints/policies/service-protection/average-latency.md]: Addressed in [8892c27](https://github.com/fluxninja/aperture/pull/2207/commits/8892c27050c63f8f9a36ccbc62c3466eceaea2b4).

File: docs/content/reference/blueprints/policies/service-protection/average-latency.md [hunk 2]
```diff
@@ -204,42 +204,42 @@
...
-    name='policy.latency_baseliner.latency_ema_limit_multiplier'
-    description='Current latency value is multiplied with this factor to calculate maximum envelope of Latency EMA.'
-    type='Number (double)'
+    name='policy.latency_baseliner.long_term_query_interval'
+    description='Interval for long-term latency query, i.e. how far back in time the query is run.'
+    type='string'
     reference=''
-    value='2'
+    value='"1800s"'
 />
```
[the remark this answers is not in the document]
Comment (hasit) [reply docs/content/reference/blueprints/policies/service-protection/average-latency.md]: Addressed in [8892c27](https://github.com/fluxninja/aperture/pull/2207/commits/8892c27050c63f8f9a36ccbc62c3466eceaea2b4)

File: docs/content/reference/blueprints/policies/service-protection/average-latency.md [hunk 3]
```diff
@@ -204,42 +204,42 @@
...
-    description='EMA parameters.'
-    type='Object (aperture.spec.v1.EMAParameters)'
-    reference='../../../spec#e-m-a-parameters'
-    value='{"correction_factor_on_max_envelope_violation": 0.95, "ema_window": "1500s", "warmup_window": "60s"}'
+    name='policy.latency_baseliner.latency_tolerance_multiplier'
+    description='Tolerance factor beyond which the service is considered to be in overloaded state. E.g. if long-term average of latency is 50ms and if the tolerance is 1.1, then the service is considered to be in an overloaded state if short-term average of latency is more than 55ms.'
+    type='Number (double)'
+    reference=''
+    value='1.25'
 />
```
[the remark this answers is not in the document]
Comment (hasit) [reply docs/content/reference/blueprints/policies/service-protection/average-latency.md]: Addressed in [8892c27](https://github.com/fluxninja/aperture/pull/2207/commits/8892c27050c63f8f9a36ccbc62c3466eceaea2b4)

File: blueprints/policies/service-protection/average-latency/policy.libsonnet [hunk 4]
```diff
@@ -14,28 +14,25 @@ function(cfg, params={}, metadata={}) {
...
+            + spec.v1.PromQL.withEvaluationInterval(evaluation_interval=updatedConfig.policy.latency_baseliner.long_term_query_periodic_interval)
+            + spec.v1.PromQL.withOutPorts({ output: spec.v1.Port.withSignalName('SIGNAL_LONG_TERM') }),
+          ),
         ),
         spec.v1.Component.withArithmeticCombinator(spec.v1.ArithmeticCombinator.mul(
-          spec.v1.Port.withSignalName('SIGNAL_EMA'),
+          spec.v1.Port.withSignalName('SIGNAL_LONG_TERM'),
           spec.v1.Port.withConstantSignal(updatedConfig.policy.latency_baseliner.latency_tolerance_multiplier),
           output=spec.v1.Port.withSignalName('SETPOINT')
         )),
```
[the remark this answers is not in the document]
Comment (hasit) [reply blueprints/policies/service-protection/average-latency/policy.libsonnet]: Addressed in [001285d](https://github.com/fluxninja/aperture/pull/2207/commits/001285d1d525ba1d17477777799584fa804494f7)

**Review by harjotgill — APPROVED** [review]

**Pull request merged**